In [2]:
import pandas as pd
from sklearn import linear_model

df= pd.read_csv("auto_data.csv")

# drop rows that contain NaN values
df = df.dropna()

df

mpg  cylinders  displacement  horsepower  weight  acceleration  \
0    18.0          8         307.0       130.0  3504.0          12.0   
1    15.0          8         350.0       165.0  3693.0          11.5   
2    18.0          8         318.0       150.0  3436.0          11.0   
3    16.0          8         304.0       150.0  3433.0          12.0   
4    17.0          8         302.0       140.0  3449.0          10.5   
..    ...        ...           ...         ...     ...           ...   
393  27.0          4         140.0        86.0  2790.0          15.6   
394  44.0          4          97.0        52.0  2130.0          24.6   
395  32.0          4         135.0        84.0  2295.0          11.6   
396  28.0          4         120.0        79.0  2625.0          18.6   
397  31.0          4         119.0        82.0  2720.0          19.4   

     model year  origin                     car name  
0            70       1  "chevrolet chevelle malibu"  
1            70       1          "buick skylark 320"  
2            70       1         "plymouth satellite"  
3            70       1              "amc rebel sst"  
4            70       1                "ford torino"  
..          ...     ...                          ...  
393          82       1            "ford mustang gl"  
394          82       2                  "vw pickup"  
395          82       1              "dodge rampage"  
396          82       1                "ford ranger"  
397          82       1                 "chevy s-10"  

[392 rows x 9 columns]

In [3]:
X = df[['weight','horsepower']]
y = df['mpg']

regr = linear_model.LinearRegression()
regr.fit(X, y)

#predict the Horsepower emission of a car where the weight is 3433kg, and the mpg is 25:
predictedhp = regr.predict([[3504, 130]])

print(predictedhp)

[19.18811123]


C:\Users\Shaun\anaconda3\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


# Coefficient

The coefficient is a factor that describes the relationship with an unknown variable.
In this case, we can ask for the coefficient value of weight against horsepower, and for mpg against horsepower. The answer(s) we get tells us what would happen if we increase, or decrease, one of the independent values.

In [4]:
print(regr.coef_)

[-0.00579416 -0.04730286]


The result array represents the coefficient values of weight and volume.

Weight: -0.00579416
Horsepower: -0.04730286

These values tell us that if the weight increases by 1kg, the mpg decreases by 0.00579416.

And if the horsepower increases by 1, the mpg decreases by 0.04730286.

What if we increase the weight with 1000kg (from 2300 to 3300) what will be the mpg?

Ans: 19.18811123 + (1000 * -0.00579416) = 13.39395123

In [6]:
predictedmpg = regr.predict([[4504, 130]])

print(predictedmpg)

[13.39395387]


C:\Users\Shaun\anaconda3\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(
